Lib/ufo2ft/outlineCompiler.py

```diff
@@ -1388,7 +1389,7 @@ def setupTable_CFF(self):
         topDict.FontBBox = self.fontBoundingBox
 
 
-class OutlineTTFCompiler(BaseOutlineCompiler):
+class OutlineTTFCompiler(BaseOutlineCompiler, InstructionCompiler):
```

Comment:
I don't like that the InstructionCompiler is used as a mixin class with the double inheritance. It would be simpler/clearer if OutlineCompiler had an InstructionCompiler.
Can the instruction compiler be used as a standalone object (without it be mixed in an outline compiler)? I don't think so because it doesn't have an `__init__` but still attempts to access self.ufo or self.otf which belong to the OutlineTTFCompiler (also see `get_instruction_compiler` in the instructionCompiler_test.py)
Now, if there's no distinction between the two and one can't be used without the other, and there's no use of having the mixin class because the way to to mix it is with the OutlineTTFCompiler class, it'd be easier to just merge them into one class, i.e. OutlienTTFCompiler grows more methods and incorporates what is now the instruction compiler.

Reply:
it doesn't even need to be an _object_, by the way, the instructionCompiler module could contain a set of functions which the OutlineTTFCompiler imports and calls to make the various tables

Reply:
What’s wrong with the mixin class?
InstructionCompiler is used as a standalone object in the tests.
I’m not a fan of a set of functions, mostly because it’s the most refactoring.

Which should it be: the current mixin, a standalone OutlineTTFCompiler.instructionCompiler, merged into OutlineTTFCompiler, or a set of functions?

Reply:
The InstructionCompiler previously was part of the postprocessing step (https://github.com/googlefonts/ufo2ft/pull/335#discussion_r714668544). As a mixin, it was easy to move it from there to the OutlineTTFCompiler without changing a lot of code.

Integrating it with OutlineTTFCompiler would more than double the number of lines in that class. So if you ask me, I’d keep it as is.

Reply:
I guess it's good to give the InstructionCompiler an __init__ method. It also makes the testing code shorter. Done in https://github.com/googlefonts/ufo2ft/pull/653/commits/60d03d028510cf9d5ec159d0228c790ec6b42c52.